in_toto/model_test.go

```diff
@@ -1128,3 +1128,54 @@ func TestValidateSupplyChainItem(t *testing.T) {
 		}
 	}
 }
+
+func TestMetablockSignWithEd25519(t *testing.T) {
+	// Test metablock signing (with ed25519)
+	// - Pass non-private key
+	// - Pass unsupported invalid key type
+	// - Pass an ed25519 key and expect a signature back
+	var key Key
+	var mb Metablock
+	mb.Load("demo.layout.template")
+
+	key.LoadPublicKey("alice.pub")
+	err := mb.Sign(key)
+	if err == nil || !strings.Contains(err.Error(), "supported yet") {
+		t.Errorf("Metablock.Sign returned (%s), expected it to claim this "+
+			"key type/scheme is unsupported", err)
+
+	}
```

Comment:
This fails because its type is non-ed25519 not because it is non-private, right?  Could you also add a test case for the latter, in order to return an error [here](https://coveralls.io/jobs/54864621/source_files/769976287#L642)?

Reply:
I did add a test [there](https://github.com/in-toto/in-toto-golang/pull/48/commits/cbe0e6106bd76fdd58d0309e28aed1029c64c8fc#diff-5b7f068c4ad310ffd7ed40b65016260eR1149), but it's unfortunately not triggered due to the order of the assertions. I may be missing something and you were thinking of a different way to trigger that line?